### PR TITLE
Manifest dry-run should be behind experiment toggle

### LINKED ISSUE #719: Problem with helm when the plugin upgrade from 2.0.3 to 2.1.0 a few days ago
I had been struggled with error message similar to following for a few days:
When we ran:
> terraform plan -var-file="vars/${ENV}.tfvars"
Following error show up, and program exit:
Error: error running dry run for a diff: template: workflow-dispatcher/templates/workflow-deployment.yaml:4:38: executing "workflow/templates/workflow-deployment.yaml" at <.Values.dispatcher.name>: nil pointer evaluating interface {}.name
  on main.tf line 57, in resource "helm_release" "walle-service":
  57: resource "helm_release" "walle-service" {

This error looks like a bug introduced by upgrading the plugin from 2.0.3 to 2.1.0. On 03/30/2021, all build succeeded with downloading helm plugin 2.0.3, but on about March 31, version 2.1.0 of helm plugin is downloaded, the error above shows.

Any idea if that is a bug, or what should I do to get the variable a value?

Thank you in advance.

### PR DESCRIPTION
**Pull request by jrhouston**
### Description

We don't want the dry run to happen unless the `manifest` experiment is switched on. 

Fixes #719 

### Release Note
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-helm/blob/main/CHANGELOG.md):

```release-note
FIX dry-run should only happen if manifest is enabled
```
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

**Review by dak1n1 — APPROVED** [review]
Looks good! Thanks for this improvement.

File: helm/resource_release.go [hunk 1]
```diff
@@ -694,107 +694,93 @@ func resourceDiff(ctx context.Context, d *schema.ResourceDiff, meta interface{})
...
-
-	if m.ExperimentEnabled("manifest") {
 		jsonManifest, err := convertYAMLManifestToJSON(dry.Manifest)
 		if err != nil {
 			return err
 		}
 		manifest := redactSensitiveValues(string(jsonManifest), d)
 		d.SetNew("manifest", manifest)
 		debug("%s set manifest: %s", logID, jsonManifest)
+	} else {
```
Comment (alexsomesan) [inline helm/resource_release.go:782]: Some would say it's considered idiomatic in Go to minimise the level of indentations and the size of the sub-branches. I think if the logic is inverted in this conditional and a return added after d.Clear("manifest") it should let the rest of the code live on the same level as it was before. Just an opinion, though.

**Review by alexsomesan — APPROVED** [review]
Looks good. I was just exploring the idea of reducing the size of the if branches, but that's only nice-to-have.

File: helm/resource_release.go (hunk 1 above)
Comment (jrhouston) [reply helm/resource_release.go:782]: I agree with this idiom in general, but as this is a feature toggle not business logic I feel like there is value having the code path behind the toggle be nested in an obvious way like this.

**Pull request merged**